kuber/observer.go

```diff
@@ -106,6 +106,8 @@ func (observer *Observer) WaitForCacheSync(timeout *time.Duration) error {
 	}
 
```

Comment:
What do you think about forcing the timeout here instead of passing the timeout period with every call?